server/src/test/java/org/elasticsearch/search/basic/SearchRedStateIndexIT.java

```diff
@@ -95,6 +99,89 @@ public void testClusterDisallowPartialsWithRedState() throws Exception {
         assertThat(ex.getDetailedMessage(), containsString("Search rejected due to missing shard"));        
     }
 
+    public void testDisallowPartialsWithRedStateRecovering() throws Exception {
+        int docCount = scaledRandomIntBetween(1000, 10000);
```

Comment:
do we need this number of docs for the test ? `10, 100` should be enough, no ?

Reply:
The test has a tendency to fail more often on larger sizes. I ended up with 10, 1000, since this makes it fail around 1/3 of the runs (but this is likely highly dependent on the hardware). Changed in: c2df1ab6996e2e35f5e7fc8bbf139fdadca09c93